feat(ui): support grid=0 in nvim_input_mouse (#32535)
neovim/neovim: runtime/doc/news.txt

• 'spellfile' location defaults to `stdpath("data").."/site/spell/"` instead of the
  first writable directoy in 'runtimepath'.
• |vim.version.range()| doesn't exclude `to` if it is equal to `from`.

> • Multigrid UIs can optionally pass 0 as the mouse grid to let Neovim

Reviewer: ```suggestion
• Multigrid UIs can optionally pass 0 as the mouse grid to let Nvim
```

• 'spellfile' location defaults to `stdpath("data").."/site/spell/"` instead of
  the first writable directory in 'runtimepath'.
• |vim.version.range()| doesn't exclude `to` if it is equal to `from`.

> • Multigrid UIs can optionally pass 0 as the mouse grid to let Nvim automatically

Reviewer 2: Let's move this up to news-features, either in UI or API section.

Author: Moved to UI

  • "Error executing Lua:" changed to "Lua:".
• 'busy' status is shown in default statusline with symbol ◐
• Improved LSP signature help rendering.

> • Multigrid UIs can optionally pass 0 as the mouse grid to let Nvim automatically
>   determine it.

Reviewer 2: does it affect things other than nvim_input_mouse ? if not, let's be explicit.
```suggestion
• Multigrid UIs can call nvim_input_mouse with grid 0 to let Nvim decide the grid.
```
